### PR TITLE
Plumb through context into sql plugin interface

### PR DESCRIPTION
**Pull request by andrewjdawson2016**
Plumbs through context timeout from managers into sql plugin interface. 

Database should enforce context timeouts. 

None, right now context.TODO() is being used.

### CODE REVIEW

File: common/persistence/sql/sqlplugin/interfaces.go [hunk 1]
```diff
@@ -503,192 +504,192 @@ type (
 
 	// tableCRUD defines the API for interacting with the database tables
 	tableCRUD interface {
```
Comment (longquanzheng) [inline common/persistence/sql/sqlplugin/interfaces.go:506]: Not a blocker: Is there something that we can remember to put this in the next release notes? This is already a public API. Any plugin implementation will get compliing errors.

**Review by longquanzheng — APPROVED** [review]

**Review by yycptt — APPROVED** [review]
LGTM

**Comment by yycptt** [comment]
For https://github.com/uber/cadence/issues/3519

File: common/persistence/sql/sqlplugin/mysql/db.go [hunk 2]
```diff
@@ -38,10 +39,10 @@ type (
...
 
 	conn interface {
-		Exec(query string, args ...interface{}) (sql.Result, error)
-		NamedExec(query string, arg interface{}) (sql.Result, error)
-		Get(dest interface{}, query string, args ...interface{}) error
-		Select(dest interface{}, query string, args ...interface{}) error
+		ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error)
+		NamedExecContext(ctx context.Context, query string, arg interface{}) (sql.Result, error)
+		GetContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
+		SelectContext(ctx context.Context, dest interface{}, query string, args ...interface{}) error
```
[the remark this answers is not in the document]
Comment (andrewjdawson2016) [reply common/persistence/sql/sqlplugin/mysql/db.go:45]: We actually have to these methods are implemented by Tx in the sqlx library. So if we want to take a context we have to use this name.

**Pull request merged**